Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -888,17 +884,33 @@ namespace Multiplayer
                 IMultiplayerSpawner* spawner = AZ::Interface<IMultiplayerSpawner>::Get();
                 if (spawner)
                 {
-                    ServerToClientConnectionData* connectionData =
-                        reinterpret_cast<ServerToClientConnectionData*>(connection->GetUserData());
-                    IReplicationWindow* replicationWindow = connectionData->GetReplicationManager().GetReplicationWindow();
-                    if (replicationWindow)
+                    // Check if this disconnected player was waiting to be spawned, and therefore, doesn't have a controlled player entity yet.
+                    bool playerSpawned = true;
+                    for (auto it = m_playersWaitingToBeSpawned.begin(); it != m_playersWaitingToBeSpawned.end(); ++it)
                     {
-                        const ReplicationSet& replicationSet = replicationWindow->GetReplicationSet();
-                        spawner->OnPlayerLeave(connectionData->GetPrimaryPlayerEntity(), replicationSet, reason);
+                        if (it->connection && it->connection->GetConnectionId() == connection->GetConnectionId())
+                        {
+                            m_playersWaitingToBeSpawned.erase(it--);
```

Comment:
That is technically an undefined behavior.
https://en.cppreference.com/w/cpp/container/vector/erase
```
Invalidates iterators and references at or after the point of the erase, including the end() iterator.


Return value
Iterator following the last removed element.

If pos refers to the last element, then the [end()](https://en.cppreference.com/w/cpp/container/vector/end) iterator is returned.

If last==end() prior to removal, then the updated [end()](https://en.cppreference.com/w/cpp/container/vector/end) iterator is returned.

If [first, last) is an empty range, then last is returned.
```

So the right pattern to use here is:
```
it = m_playersWaitingToBeSpawned.erase(it);
```

Reply:
thank you! fixed!